### PR TITLE
fix: Add `tippy.js` as a direct dependency

### LINKED ISSUE #856: Display the ComboBox dropdown (from a Dialog) inside AppUI is throwing an exception
We can run the following [sandbox ](https://codesandbox.io/s/itwinui-react-minimal-example-forked-0phivr?file=/src/App.tsx:557-1010) and the ComboBox dropdown works as expected.

But if we copy the identical dialog to ninezone (Design Review), and try to display the ComboBox dropdown, we get the following error:

image

image

We are using:
 "@itwin/itwinui-react": "1.46.0"

### PR DESCRIPTION
**Pull request by MindaugasButkus**
tippyjs version 6.3.1 had some bugs in it that are fixed in version 6.3.7. One of which was clicking on a Dropdown would cause an error because "props.onShow is not a function". Upgrading to 6.3.7 resolves the issue.

Closes # <no issue was created>

## Checklist

- [ ] Add meaningful unit tests for your component (verify that all lines are covered) - Code was not modified
- [x] Verify that all existing tests pass
- [ ] Add component features demo in Storybook (different stories) - No new components were added
- [ ] Approve test images for new stories - No new stories were added
- [ ] Add screenshots of the key elements of the component - No components were modified

### CODE REVIEW

**Review by mayank99 — COMMENTED** [review]
Hi @MindaugasButkus, thanks for the PR. Can I ask why you need this change to be made in iTwinUI-react?

We only have a dependency on `@tippyjs/react@^4.2.5`. The `^` means that 4.2.6 is allowed. And it depends on `tippy.js@^6.3.1`, which again means that 6.3.7 is compatible. So if you run `npm install`, it should automatically install those versions as they are the latest. If your lockfile already had the old versions, then you can use [`npm update`](https://docs.npmjs.com/cli/v8/commands/npm-update) on these two packages.

**Comment by MindaugasButkus** [comment]
> Hi @MindaugasButkus, thanks for the PR. Can I ask why you need this change to be made in iTwinUI-react?
> 
> We only have a dependency on `@tippyjs/react@^4.2.5`. The `^` means that 4.2.6 is allowed. And it depends on `tippy.js@^6.3.1`, which again means that 6.3.7 is compatible. So if you run `npm install`, it should automatically install those versions as they are the latest. If your lockfile already had the old versions, then you can use [`npm update`](https://docs.npmjs.com/cli/v8/commands/npm-update) on these two packages.

The problem was that the consumer of this package was using yarn, not npm, and there is no easy way to upgrade indirect dependencies in yarn.

Now, with this change there should be a version bump in this package, so it is a direct dependency update. With this direct dependency update comes the requirement for newer versions of tippy, which I added to `dependencies` list.

**Comment by mayank99** [comment]
> Hi @mayank99. The problem was that the consumer of this package was using yarn, not npm, and there is no easy way to upgrade indirect dependencies in yarn.

There is a [`yarn upgrade`](https://classic.yarnpkg.com/lang/en/docs/cli/upgrade/) command too

> Now, with this change there should be a version bump in this package, so it is a direct dependency update. With this direct dependency update comes the requirement for newer versions of tippy, which I added to `dependencies` list.

But we don't really have a direct runtime dependency on tippy.js, we only use types from there. Theoretically, if `@tippyjs/react` was to remove tippy.js as a dependency in a future minor version, our package should still work.

Am I missing something? @bentleyvk

**Review by mayank99 — APPROVED** [review]
If it helps people, let's go with it I guess. I don't see much of a downside, as `tippy.js` will be included one way or another.

**Review by adamhock — APPROVED** [review]

File: packages/iTwinUI-react/package.json
Comment (rohan-kadkol) [inline packages/iTwinUI-react/package.json:50]: Just curious, is there a reason (e.g. fixes something) for making this upgrade to tippyjs/react? Or would just upgrading tippy.js (not react) from "6.3.1 -> 6.3.7" fix the problem you were mentioning?

**Review by r100-stack — COMMENTED** [review]
Just reporting what I saw. I reconfirmed that tippyjs react was using version "^6.3.1", and that `yarn upgrade` does not upgrade indirect dependencies. But I wasn't sure how to confirm if tippyjs's version "6.3.7" fixes the "props.onShow is not a function" you are mentioning.

I also tried looking for uses of tippy.js in our repo, there were very few. Since it is not a major upgrade of the tippy.js (x.x.1 -> x.x.7) and since all our tests pass, I think this update might be alright.

But probably better to wait for a review/approval from Greta or Adam.

File: packages/iTwinUI-react/package.json
Comment (MindaugasButkus) [reply packages/iTwinUI-react/package.json:50]: @rohan-kadkol there is some kind of fix for dynamic singleton headless content: https://github.com/atomiks/tippyjs-react/commit/d4c915de38b246d87abfad576887063d0a09d463 . It is unrelated, but I see no reason keeping the outdated version. It's a related package, so why not upgrade?

Regarding tippy.js package, I think the issue was fixed in version 6.3.3 with this commit: https://github.com/atomiks/tippyjs/commit/22481d7655b6eed7884ce84daef52471d526d1dc

**Comment by MindaugasButkus** [comment]
Could anyone tell me what the "Project / Add to project (pull_request)" check does and why did it fail? Can the PR be completed without it passing? If no, how can I rerun it?

**Comment by gretanausedaite** [comment]
> Could anyone tell me what the "Project / Add to project (pull_request)" check does and why did it fail? Can the PR be completed without it passing? If no, how can I rerun it?

This action is for our internal task management and is not required, PR can be merged without this.

> Also, I do not have the required permissions to merge this PR. @gretanausedaite @adamhock Could you merge it?

Sure, doing it now.

**Pull request merged**
